src/main/java/cody/TaskManager.java

```diff
@@ -0,0 +1,208 @@
+package cody;
+
+import cody.tasks.Deadline;
+import cody.tasks.Event;
+import cody.tasks.Task;
+import cody.tasks.Todo;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileNotFoundException;
+
+
+public class TaskManager {
+    ArrayList<Task> tasks;
+    private static final String BORDER = "______________________________________________________________\n";
+
+    private void printList() {
+        System.out.print(BORDER + " Here are the tasks in your list:\n");
+        for (int i = 0; i < tasks.size(); i++) {
+            Task task = tasks.get(i);
+            System.out.println(" " + (i + 1) + ". [" + task.getTaskType() + "] "
+                    + "[" + task.getStatusIcon() + "] "
+                    + task.getDescription());
+        }
+        System.out.print(BORDER);
+    }
+
+    private void markTask(int index) {
+        System.out.print(BORDER + " Good job! I've marked this task as done:\n"
+                + " [" + tasks.get(index).getStatusIcon() + "] " + tasks.get(index).getDescription() + "\n"
+                + BORDER);
+    }
+
+    private void handleMarking(String input) {
+        try {
+            String[] parts = input.split(" ");
+            int index = Integer.parseInt(parts[1]) - 1;
+            boolean isDone = input.startsWith("mark");
+            tasks.get(index).markTask(isDone);
+            markTask(index);
+        } catch (NumberFormatException e) {
+            System.out.print(BORDER + " Task number is invalid\n"
+                    + " Please enter a valid number\n" + BORDER);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.print(BORDER + " Task number is out of range\n"
+                    + " Please enter a number between 1 and " + tasks.size() + "\n" + BORDER);
+        }
+    }
+
+    private void addTask(String input) {
+        try {
+            Task task = createTaskFromInput(input);
+            tasks.add(task);
+            printTask(task);
+        } catch (CodyException e) {
+            System.out.print(BORDER + e.getMessage() + BORDER);
+        }
+    }
+
+    private Task createTaskFromInput(String input) throws CodyException {
+        if (input.startsWith("todo")) {
+            return createTodoTask(input);
+        } else if (input.startsWith("deadline")) {
+            return createDeadlineTask(input);
+        } else if (input.startsWith("event")) {
+            return createEventTask(input);
+        } else {
+            // If the task type is unknown, throw a CodyException
+            throw new CodyException(" I'm not sure what task this is\n"
+                    + " Please start with 'todo', 'event' or 'deadline'\n");
+        }
+    }
+
+
+    // Creates a cody.tasks.Todo task from the input.
```

Comment:
```suggestion
    // Creates a Todo task from the input.
```
There is no need to specify the exact package path since it is easy to tell what Todo is, please check the other comments including this as well